src/openforms/config/views.py

```diff
@@ -1,48 +1,21 @@
-from typing import Any, Generator, Protocol, TypeGuard
+from typing import Any
 
 from django.contrib.auth.mixins import PermissionRequiredMixin
 from django.urls import reverse
-from django.utils.encoding import force_str
 from django.utils.translation import gettext as _
 from django.views.generic import TemplateView
 
-from openforms.appointments.registry import register as appointments_register
-from openforms.contrib.brk.checks import BRKValidatorCheck
-from openforms.contrib.kadaster.config_check import BAGCheck, LocatieServerCheck
-from openforms.contrib.kvk.checks import KVKRemoteValidatorCheck
-from openforms.dmn.registry import register as dmn_register
-from openforms.payments.registry import register as payments_register
-from openforms.plugins.plugin import AbstractBasePlugin
-from openforms.prefill.registry import register as prefill_register
-from openforms.registrations.registry import register as registrations_register
 from openforms.utils.mixins import UserIsStaffMixin
 
-from .data import Action, Entry
+from .checks import ConfigurationCheck
+from .data import Entry
 from .models import GlobalConfiguration
 from .utils import verify_clamav_connection
 
 
-def _subset_match(requested: str | None, checking: str) -> bool:
-    if not requested:
-        return True
-    return requested == checking
-
-
-class ConfigCheckable(Protocol):
-    verbose_name: str
-
-    def check_config(self) -> None: ...
-
-    def get_config_actions(self) -> list[Action]: ...
-
-
-def is_plugin(plugin: Any) -> TypeGuard[AbstractBasePlugin]:
-    if hasattr(plugin, "identifier"):
-        return True
-    return False
-
-
-class ConfigurationView(UserIsStaffMixin, PermissionRequiredMixin, TemplateView):
+class ConfigurationView(
+    UserIsStaffMixin, PermissionRequiredMixin, ConfigurationCheck, TemplateView
+):
```

Comment:
I would not add `ConfigurationCheck` as a mixin, can't you just instantiate an instance in the view and use that?

```py
def get_context_data(self, ...):
    ...

    check = ConfigurationCheck(requested_plugin=self.request.GET.get("plugin"))
    sections = check.get_configuration_results(module)
    ...
```
the `ConfigurationCheck` class is completely decoupled from the request/response/view, so there's really no reason to not do simple composition here.